### DISCUSSION TITLE
How to make responsive a Masonry page?

### DISCUSSION THREAD

**Discussion by fmarzocca** [quasarframework/quasar#13776] [General - Components / Directives / etc] [answered]
Hi,
I am working at a layout as the Masonry example on the Docs: https://quasar.dev/layout/grid/flexbox-patterns#example--masonry

Unfortunately, the example is not responsive. I am struggling to make it run also on small screens (showing each div at full width), but I am hanging...
Is there any doc I can read for this?

**Comment by MilosPaunovic** [comment]
You should create a Codepen example of what you have now, so someone could take a deeper dive into it and provide support.

**Comment by fmarzocca** [comment]
@MilosPaunovic, yes you're right, sorry.

I made a Codepen [here](https://codepen.io/fmarzocca/pen/RwQXvRd).
As you can see it is responsive. Being the columns with different heights, in full page it renders bad. I'd like to have a Masonry view in full page, but keep the responsive setting.

**Comment by fmarzocca** [comment]
still no success....

**Comment by TobyMosque** [comment, marked as answer]
https://codepen.io/tobias-mesquita/pen/bGvbKGP

you don't need to define $breakpoint-{size} and $sizes, they are quasar variables

**Reply by fmarzocca** [reply]
@TobyMosque , if I may ask... what is this for?

```
.example-container
      height: calc(2800px / #{$x}) 
```

it produces a long white space on smartphone if I have just 4 or 5 cards on the page... Is it a sort of placeholder?

**Reply by TobyMosque** [reply]
the original codepen had a fixed height of 700px, i just make that semi-dynamic

**Reply by fmarzocca** [reply]
Understand. I am trying to reduce that space, as on smartphone devices it is a very long blank space after the latest cell. That `height` depends upon how many cells (and what is the cell height). If you have just 5 or 6 cells, that blank space is very long. Do you have any clue?
